src/network/url_downloader.cpp

```diff
@@ -102,7 +102,21 @@ download(QNetworkAccessManager* manager, const Time& timeout, QUrl const& url, P
 
     if (reply->error() != QNetworkReply::NoError)
     {
-        const auto msg = download_timeout.isActive() ? reply->errorString().toStdString() : "Network timeout";
+        const auto error_code = reply->error();
+        const auto error_string = reply->errorString().toStdString();
+
+        // Log the original error message at debug level
+        mpl::log(mpl::Level::debug,
+                 category,
+                 fmt::format("Qt error {}: {}", static_cast<int>(error_code), error_string));
+
+        const bool is_timeout_error = (error_code == QNetworkReply::TimeoutError);
+        const auto msg = is_timeout_error ? "Network timeout" : error_string;
+
+        if (is_timeout_error && download_timeout.isActive())
+        {
+            mpl::log(mpl::Level::debug, category, "Timeout error detected but download_timeout is still active");
```

Comment:
Should this be a warning though?

Reply:
> Should this be a warning though?
I guess it can be at error level, but this should never be triggered by an OperationCanceledError and doesn't have any impact on the rest of the error handling which is why I originally put it as debug